pkg/kubelet/cm/devicemanager/topology_hints.go

```diff
@@ -230,70 +232,14 @@ func (m *ManagerImpl) getNUMANodeIds(topology *pluginapi.TopologyInfo) []int {
 }
 
 func (m *ManagerImpl) getPodDeviceRequest(pod *v1.Pod) map[string]int {
-	podResources := sets.NewString()
-
-	// Find the max request of a given resource across all init containers
-	initContainerRequests := make(map[string]int)
-	for _, container := range pod.Spec.InitContainers {
-		for resourceObj, requestedObj := range container.Resources.Limits {
-			resource := string(resourceObj)
-			requested := int(requestedObj.Value())
-
-			if !m.isDevicePluginResource(resource) {
-				continue
-			}
-
-			podResources.Insert(resource)
-
-			if _, exists := initContainerRequests[resource]; !exists {
-				initContainerRequests[resource] = requested
-				continue
-			}
-			if requested > initContainerRequests[resource] {
-				initContainerRequests[resource] = requested
-
-			}
-		}
-	}
-
-	// Compute the sum of requests across all app containers for a given resource
-	appContainerRequests := make(map[string]int)
-	for _, container := range pod.Spec.Containers {
-		for resourceObj, requestedObj := range container.Resources.Limits {
-			resource := string(resourceObj)
-			requested := int(requestedObj.Value())
-
-			if !m.isDevicePluginResource(resource) {
-				continue
-			}
-			podResources.Insert(resource)
-			appContainerRequests[resource] += requested
-		}
-	}
-
-	// Calculate podRequests as the max of init and app container requests for a given resource
+	// for these device plugin resources, requests == limits
+	limits := corev1helpers.PodLimits(pod, nil)
```

Comment:
There's a slight change in functionality here in that it also includes the pod overhead in the calculation now. It seemed more correct that way, but it's easy to avoid it if needed.

Reply:
cc: @ffromani @swatisehgal what do you think about including pod overhead here?

Reply:
My understanding was that pod overhead was used to specify CPU and memory that the container(s) would need to run. I haven't come across a scenario where devices are specified as pod overhead as typically there are limited devices on the node and they are expensive . It might be worth checking if it is valid to specify devices as overhead in the runtime class.  If we support that, it makes sense to honor the request and consider it while accounting.

Reply:
In the deleted code we consider only the resources provided by device plugins:
```
if !m.isDevicePluginResource(resource) {
     continue
}
```
I welcome cleanups, this code could use quite some improvements, but I think it should be more evident here we only care about devices.

Reply:
(deleted a previous comment which was inaccurate)
I also remember that overhead is only about cpus and memory which we will ignore just before: this code only wants to deal with devices. While the current change work, I'd explicitely call the version without overhead.

On top of that, I can't help but wonder: this is pretty much a special case of accumulating a subset of resources. I'm not sure it fits cleanly enough in the scope of this cleanup.

Reply:
I'll update to exclude overhead. The primary purpose of this PR isn't really just the de-duplication, but to avoid a set of subtle bugs in the future with respect to this calculation.  E.g. When sidecar containers are implemented, that PR would need to update this code as well to change the calculation to incorporate sidecars (init containers that don't terminate).  Otherwise sidecars which request a device, though perhaps unlikely to occur in practice, would make this calculation incorrect.

Reply:
@tzneal fair point and this addresses my concern